src/serviceworker/sw.js

```diff
@@ -101,6 +131,18 @@ export class ServiceWorker {
 		           })
 	}
 
+	async fromCacheOrFetchAndCache(request: Request): Promise<Response> {
+		const cache = await this._caches.open(this._cacheName)
+		const response = await cache.match(request.url)
+		if (response) {
+			return response
+		} else {
+			const networkResponse = await fetch(request)
+			await cache.put(request, networkResponse.clone())
+			return networkResponse
+		}
+	}
+
```

Comment:
async await in service worker. we should change this or not include in this release